var/spack/repos/builtin/packages/uncrustify/package.py

```diff
@@ -31,28 +31,15 @@ class Uncrustify(Package):
     version("0.62", commit="5987f2")
     version("0.61", sha256="1df0e5a2716e256f0a4993db12f23d10195b3030326fdf2e07f8e6421e172df9")
 
-    depends_on("cmake", type="build", when="@0.64:")
-    depends_on("automake", type="build", when="@0.63")
-    depends_on("autoconf", type="build", when="@0.63")
-
-    @when("@0.64:")
-    def install(self, spec, prefix):
-        with working_dir("spack-build", create=True):
-            cmake("..", *std_cmake_args)
-            make()
-            make("install")
-
-    @when("@0.63")
-    def install(self, spec, prefix):
-        which("bash")("autogen.sh")
-        configure("--prefix={0}".format(self.prefix))
-        make()
-        make("install")
-
-    @when("@:0.62")
-    def install(self, spec, prefix):
-        configure("--prefix={0}".format(self.prefix))
-        make()
-        make("install")
+    buildsystem(
+        conditional("cmake", when="@0.64:"),
+        conditional("autotools", when="@:0.63"),
+        default="cmake",
+    )
```

Comment:
I still don't love this syntax. I would prefer something like:
```python
build_system("cmake", when="@0.64:")
build_system("autotolls", when="@:0.63")
```
This matches how most other directives work.

Reply:
It's a variant, so it's difficult to do as you say, since we'd need a default value shared among 2 directives. Also, `arpack-ng` would read:
```python
buildsystem("cmake")
buildsystem("autotools")
```
With that it is imo less clear that you can have only one at a time.